code/datums/components/surgery_initiator.dm

```diff
@@ -0,0 +1,281 @@
+/// Allows an item to be used to initiate surgeries.
+/datum/component/surgery_initiator
+	/// If present, this surgery TYPE will be attempted when the item is used.
+	/// Useful for things like limb reattachments that don't need a scalpel.
+	var/datum/surgery/forced_surgery
+
+	/// If true, the initial step will be cancellable by just using the tool again. Should be FALSE for any tool that actually has a first surgery step.
+	var/can_cancel_before_first = FALSE
+
+	/// If true, can be used with a cautery in the off-hand to cancel a surgery.
+	var/can_cancel = TRUE
+
+/**
+ * Create a new surgery initiating component.
+ *
+ * Arguments:
+ * * forced_surgery - (optional) the surgery that will be started when the parent is used on a mob.
+ */
+/datum/component/surgery_initiator/Initialize(datum/surgery/forced_surgery)
+	. = ..()
+	if(!isitem(parent))
+		return COMPONENT_INCOMPATIBLE
+
+	src.forced_surgery = forced_surgery
+
+/datum/component/surgery_initiator/Destroy(force, silent)
+	return ..()
+
+/datum/component/surgery_initiator/RegisterWithParent()
+	RegisterSignal(parent, COMSIG_ITEM_ATTACK, .proc/initiate_surgery_moment)
+	RegisterSignal(parent, COMSIG_ATOM_UPDATE_SHARPNESS, .proc/on_parent_sharpness_change)
+
+/datum/component/surgery_initiator/UnregisterFromParent()
+	UnregisterSignal(parent, COMSIG_ITEM_ATTACK)
+	UnregisterSignal(parent, COMSIG_ATOM_UPDATE_SHARPNESS)
+
+/// Keep tabs on the attached item's sharpness. Any sharp item can start a surgery.
+/datum/component/surgery_initiator/proc/on_parent_sharpness_change()
+	SIGNAL_HANDLER  // COMSIG_ATOM_UPDATE_SHARPNESS
+	var/obj/item/P = parent
+	if(!P.sharp)
+		RemoveComponent()
+
+/// Does the surgery initiation.
+/datum/component/surgery_initiator/proc/initiate_surgery_moment(datum/source, atom/target, mob/user)
+	SIGNAL_HANDLER	// COMSIG_ITEM_ATTACK
+	if(!isliving(user))
+		return
+	if(!user.Adjacent(target))
+		return
+	if(user.a_intent != INTENT_HELP)
+		return
+	var/mob/living/L = user
+	if(istype(L) && L.has_status_effect(STATUS_EFFECT_SUMMONEDGHOST))
+		to_chat(user, "<span class='notice'>You realise that a ghost probably doesn't have any useful organs.</span>")
+		return //no cult ghost surgery please
+	INVOKE_ASYNC(src, .proc/do_initiate_surgery_moment, target, user)
+	return TRUE
```

Comment:
if it returns true shouldnt the failures be false? is the return value ever used?

Reply:
Nah, this actually feeds back into the attack chain. If it returns false here (or nothing at all), it attacks the person with the tool before popping up the surgery dialog.